### PR TITLE
[13.0] [MIG] sale_timesheet_line_exclude

### PR DESCRIPTION
**Pull request by Du-ma**
Migration

### CODE REVIEW

File: sale_timesheet_line_exclude/models/account_analytic_line.py [hunk 1]
```diff
@@ -29,26 +29,24 @@ def _onchange_exclude_from_sale_order(self):
...
     def _constrains_exclude_from_sale_order(self):
-        if self.filtered(
-            lambda line: line.timesheet_invoice_id
-            and line.so_line.product_id.invoice_policy == "delivery"
-        ):
-            raise ValidationError(
-                _(
-                    "You can not modify timesheets in a way that would affect"
-                    " invoices since these timesheets were already invoiced."
+        for line in self:
```
Comment (alexey-pelykh) [inline sale_timesheet_line_exclude/models/account_analytic_line.py:32]: Optimization?

File: sale_timesheet_line_exclude/models/account_analytic_line.py [hunk 2]
```diff
@@ -68,48 +66,37 @@ def _timesheet_should_evaluate_so_line(self, values, check):
...
-            else self.task_id,
-            "employee": (self.env["hr.employee"].sudo().browse(values["employee_id"]))
-            if values and "employee_id" in values
-            else self.employee_id,
-        }
+        if values:
+            return {
+                "task": self.env["project.task"].sudo().browse(values.get("task_id")),
+                "employee": self.env["hr.employee"]
+                .sudo()
```
Comment (alexey-pelykh) [inline sale_timesheet_line_exclude/models/account_analytic_line.py:74]: sudo on next line, ugh

File: sale_timesheet_line_exclude/models/account_analytic_line.py (hunk 1 above)
Comment (Du-ma) [reply sale_timesheet_line_exclude/models/account_analytic_line.py:32]: Yup, removed one for loop :grin:

File: sale_timesheet_line_exclude/models/account_analytic_line.py (hunk 2 above)
Comment (Du-ma) [reply sale_timesheet_line_exclude/models/account_analytic_line.py:74]: Can't really do anything about it, `pre-commit` made it "PrEtTy" :smile: 
If it's that much of an eyesore i can make a variable referencing sudo employee obj :smile:

File: sale_timesheet_line_exclude/models/account_analytic_line.py (hunk 1 above)
Comment (Du-ma) [reply sale_timesheet_line_exclude/models/account_analytic_line.py:32]: Whoops, I mean skipping other elements instead of continuing the for loop

**Comment by gaikaz** [comment]
@alexey-pelykh Could you re-review?

**Review by dsolanki-initos — APPROVED** [review]
LGTM

**Review by alexey-pelykh — APPROVED** [review]

**Review by gaikaz — APPROVED** [review]

**Pull request merged**